### PR TITLE
Added sqlclosecheck to linters

### PR DESCRIPTION
**Pull request by ryanrolds**
Adds [sqlclosecheck](https://github.com/ryanrolds/sqlclosecheck) linter

The goal of the linter is to identify Rows/Stmt that are not closed or closed using a defer.

### CODE REVIEW

File: test/testdata/sqlclosecheck.go [hunk 1]
```diff
@@ -0,0 +1,353 @@
+//args: -Esqlclosecheck
+package testdata
```
Comment (sayboras) [inline test/testdata/sqlclosecheck.go:2]: very extensive testing :tada:

File: test/testdata/sqlclosecheck.go [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+)
+
+var (
+	ctx context.Context
+	db  *sql.DB
+	dbx *sqlx.DB
+)
+
+func rowsCorrectDeferBlock() {
+	age := 27
```
Comment (daixiang0) [inline test/testdata/sqlclosecheck.go:21]: Could we use const num here?

File: go.sum [hunk 3]
```diff
@@ -314,6 +316,10 @@ github.com/rogpeppe/go-internal v1.5.2/go.mod h1:xXDCJY+GAPziupqXw64V24skbSoqbTE
 github.com/russross/blackfriday/v2 v2.0.1/go.mod h1:+Rmxgy9KzJVeS9/2gXHxylqXiyQDYRxCVz55jmeOWTM=
 github.com/ryancurrah/gomodguard v1.1.0 h1:DWbye9KyMgytn8uYpuHkwf0RHqAYO6Ay/D0TbCpPtVU=
 github.com/ryancurrah/gomodguard v1.1.0/go.mod h1:4O8tr7hBODaGE6VIhfJDHcwzh5GUccKSJBU0UMXJFVM=
+github.com/ryanrolds/sqlclosecheck v0.2.0 h1:lzXPc4X12yhUqHHU2PWl45YGbV4jpCmhcrvgLxlZDeg=
```
Comment (sayboras) [inline go.sum:319]: not sure why this entry is here as the version in go.mod is v0.3.0-rc. Can you help to run _go mod tidy_ once ?

Comment (ryanrolds) [reply go.sum:319]: I will fix that.

File: test/testdata/sqlclosecheck.go (hunk 2 above)
Comment (ryanrolds) [reply test/testdata/sqlclosecheck.go:21]: Will do.

File: go.sum (hunk 3 above)
Comment (ryanrolds) [reply go.sum:319]: I update the version of `sqlclosecheck` while I was working on the PR and didn't run tidy.

**Comment by ryanrolds** [comment]
All feedback should be addressed. Thank you for taking the time to review.

File: go.mod [hunk 4]
```diff
@@ -37,6 +37,7 @@ require (
 	github.com/nishanths/exhaustive v0.0.0-20200525081945-8e46705b6132
 	github.com/pkg/errors v0.9.1
 	github.com/ryancurrah/gomodguard v1.1.0
+	github.com/ryanrolds/sqlclosecheck v0.3.0-rc
```
Comment (daixiang0) [inline go.mod:40]: could we pick a release version not a rc version?

Comment (ryanrolds) [reply go.mod]: Yup. Set to v0.3.0 now.

**Comment by ryanrolds** [comment]
Anything I need to do at this point? What's the process for merging and releasing?

**Pull request merged**